purge-docker-cluster.yml

```diff
@@ -1,43 +1,447 @@
+
+  vars_prompt:
+    - name: ireallymeanit
+      prompt: Are you sure you want to purge the cluster?
+      default: 'no'
+      private: no
+
+  tasks:
+  - name: exit playbook, if user did not mean to purge cluster
+    fail:
+      msg: >
+        "Exiting purge-docker-cluster playbook, cluster was NOT purged.
+         To purge the cluster, either say 'yes' on the prompt or
+         or use `-e ireallymeanit=yes` on the command line when
+         invoking the playbook"
+    when: ireallymeanit != 'yes'
+
+
+- name: retrieve variables
+
+  hosts:
+    - all
+    - localhost
+
+  tasks:
+    - include_vars: roles/ceph-common/defaults/main.yml
+    - include_vars: roles/ceph-mon/defaults/main.yml
+    - include_vars: roles/ceph-osd/defaults/main.yml
+    - include_vars: roles/ceph-mds/defaults/main.yml
+    - include_vars: roles/ceph-rgw/defaults/main.yml
+    - include_vars: roles/ceph-nfs/defaults/main.yml
+    - include_vars: roles/ceph-restapi/defaults/main.yml
+    - include_vars: group_vars/all
+
+    - name: check if it is Atomic host
+      stat: path=/run/ostree-booted
+      register: stat_ostree
+
+    - name: set fact for using Atomic host
+      set_fact:
+        is_atomic: "{{ stat_ostree.stat.exists }}"
+
+
+- name: purge ceph mds cluster
+
+  hosts:
+    - mdss
+
+  become: true
 
   tasks:
+  - name: disable ceph mds service
+    service:
+      name: "ceph-mds@{{ ansible_hostname }}"
+      state: stopped
+      enabled: no
+    ignore_errors: true
+
+  - name: remove ceph mds container
+    docker:
+      image: "{{ ceph_mds_docker_username }}/{{ ceph_mds_docker_imagename }}:{{ ceph_mds_docker_image_tag }}"
+      name: "{{ ansible_hostname }}"
+      state: absent
+    ignore_errors: true
+
+  - name: remove ceph mds service
+    file:
+        path: /etc/systemd/system/ceph-mds@.service
+        state: absent
+
+  - name: remove ceph mds image
+    docker_image:
+      state: absent
+      name: "{{ ceph_mds_docker_username }}/{{ ceph_mds_docker_imagename }}"
+      tag: "{{ ceph_mds_docker_image_tag }}"
+    tags:
+      remove_img
+
+- name: purge ceph rgw cluster
+
+  hosts:
+    - rgws
+
+  become: true
+
+  tasks:
+  - name: disable ceph rgw service
+    service:
+      name: "ceph-rgw@{{ ansible_hostname }}"
+      state: stopped
+      enabled: no
+    ignore_errors: true
+
+  - name: remove ceph rgw container
+    docker:
+      image: "{{ ceph_rgw_docker_username }}/{{ ceph_rgw_docker_imagename }}:{{ ceph_rgw_docker_image_tag }}"
+      name: "{{ ansible_hostname }}"
+      state: absent
+    ignore_errors: true
+
+  - name: remove ceph rgw service
+    file:
+        path: /etc/systemd/system/ceph-rgw@.service
+        state: absent
+
+  - name: remove ceph rgw image
+    docker_image:
+      state: absent
+      name: "{{ ceph_rgw_docker_username }}/{{ ceph_rgw_docker_imagename }}"
+      tag: "{{ ceph_rgw_docker_image_tag }}"
+    tags:
+      remove_img
+
+  - name: remove libnss3-tools on redhat
```

Comment:
I wasn't sure exactly. It's just that in https://github.com/ceph/ceph-ansible/blob/master/roles/ceph-rgw/tasks/main.yml#L5 we `include: openstack-keystone.yml` based on this `radosgw_keystone` conditional variable. If this isn't container specific we should probably have an additional `not rgw_containerized_deployment` check and I can remove it from this playbook.